src/bin/cql-stress-cassandra-stress/settings/option/mode.rs

```diff
@@ -0,0 +1,212 @@
+        parser.print_help();
+    }
+
+    pub fn print_settings(&self) {
+        println!("Mode:");
+        println!("  Compression: {:?}", self.compression);
+        if let Some((username, password)) = &self.user_credentials {
+            println!("  Username: {}", username);
+            println!("  Password: {}", password);
+        }
+        println!("  Pool size: {:?}", self.pool_size);
+    }
+
+    fn from_handles(handles: ModeParamHandles) -> Result<ModeOption> {
+        let compression = handles.compression.get().unwrap();
+        let user_credentials = match (handles.username.get(), handles.password.get()) {
+            (Some(username), Some(password)) => Some((username, password)),
+            (None, None) => None,
+            _ => {
+                return Err(anyhow::anyhow!(
+                    "Both username and password need to be either set or unspecified"
+                ))
+            }
+        };
+        let pool_size = match handles.connections_per_shard.get() {
+            Some(per_shard) => per_shard,
+            None => handles.connections_per_host.get().unwrap(),
+        };
+
+        Ok(Self {
+            compression,
+            user_credentials,
+            pool_size,
+        })
+    }
+}
+
+struct ModeParamHandles {
+    compression: SimpleParamHandle<Option<Compression>>,
+    username: SimpleParamHandle<String>,
+    password: SimpleParamHandle<String>,
+    connections_per_host: SimpleParamHandle<ConnectionsPerHost>,
+    connections_per_shard: SimpleParamHandle<ConnectionsPerShard>,
+}
+
+fn prepare_parser() -> (ParamsParser, ModeParamHandles) {
+    let mut parser = ParamsParser::new(ModeOption::CLI_STRING);
+
+    // `cql3` and `native` parameters are ignored but we introduce them so we support
+    // the cassandra-stress CLI syntax which is used in SCT.
+    let _cql3 = parser.simple_param::<bool>("cql3", None, "cql3 mode (dummy parameter)", false);
+    let _native =
+        parser.simple_param::<bool>("native", None, "native mode (dummy parameter)", false);
+
+    let compression = parser.simple_param(
+        "compression=",
+        Some("none"),
+        "Compression algorithm used for connections",
+        false,
+    );
+    let username = parser.simple_param("user=", None, "username", false);
+    let password = parser.simple_param("password=", None, "password", false);
+    let connections_per_shard = parser.simple_param(
+        "connectionsPerShard=",
+        Some("1"),
+        "Number of connections per shard",
+        false,
+    );
+    let connections_per_host = parser.simple_param(
+        "connectionsPerHost=",
+        None,
+        "Number of connections per host",
+        false,
+    );
+
+    // $ ./cql-stress-cassandra-stress help -node
+    // Usage: -mode cql3 native [compression=?] [user=?] [password=?] [connectionsPerShard=?]
+    //  OR
+    // Usage: -mode cql3 native [compression=?] [user=?] [password=?] [connectionsPerHost=?]
```

Comment:
If we add more mutually exclusive groups of parameters like `connectionsPerHost` + `connectionsPerShar` in the future, we might get a combinatorial explosion with this approach.

Just pointing that out as a potential problem in the future, let's not worry about it too much right now.

Reply:
Yeah, I agree - I think the parameters' grouping should be somehow refactored in the future. As of now it's not very convenient to work with the parameter groups but at least it works..